components/brave_wallet/common/mem_utils.h

```diff
@@ -31,7 +33,8 @@ struct SecureZeroAllocator {
     return static_cast<T*>(::operator new(n * sizeof(T)));
   }
   void deallocate(T* p, size_t n) {
-    SecureZeroData(p, n);
+    SecureZeroData(
+        base::as_writable_bytes(UNSAFE_BUFFERS(base::make_span(p, n))));
```

Comment:
do we need need `UNSAFE_BUFFERS` as `make_span` already return one with `UNSAFE_BUFFERS`?

Reply:
Unfortunately yes, otherwise this line returns an error when building with `-Wunsafe-buffer-usage`. This is the case whenever instantiating a span with an unbounded pointer/size pair.

Reply:
Did we have a potential bug there? It was allocating `n * sizeof(T)` bytes, but clearing only first `n` bytes. Works correctly only when `sizeof(T) == 1` which seems the only way it was instantiated with.